friends/src/addFriend.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+
+export default class Add extends React.Component {
+
+  constructor(props) {
+    super(props)
+    
+    this.state ={
+      name: '',
+      age: '',
+      email: '',
+      id: ''
+    } 
+  }
+  
+  changeHandler = (e) => {
+    this.setState({[e.target.name]: e.target.value, id: this.props.friends.length + 1})
```

Comment:
Once you make that change then you only have to focus on updating the inputs 
```suggestion
    this.setState({[e.target.name]: e.target.value})
```